lib/puppetserver/ca/local_certificate_authority.rb

```diff
@@ -73,33 +82,60 @@ def create_master_cert(ca_key, ca_cert)
                                               @settings[:hostpubkey])
         if master_key
           master_csr = @host.create_csr(name: @settings[:certname], key: master_key)
-          master_cert = sign_master_cert(ca_key, ca_cert, master_csr)
+          if @settings[:subject_alt_names].empty?
+            alt_names = "DNS:puppet, DNS:#{@settings[:certname]}"
+          else
+            alt_names = @settings[:subject_alt_names]
+          end
+
+          master_cert = sign_authorized_cert(ca_key, ca_cert, master_csr, alt_names)
         end
 
         return master_key, master_cert
       end
 
-      def sign_master_cert(int_key, int_cert, csr)
+      # Used when generating certificates offline.
+      def load_ca
+        if File.exist?(@settings[:cacert]) && File.exist?(@settings[:cakey]) && File.exist?(@settings[:cacrl])
+          signing_cert, key = nil
```

Comment:
This kinda works, but not explicitly, iirc. I think that any additional variables in a multi-assignment line get `nil`, so like
```
a, b, c = 1, 2
```
will leave `a == 1`, `b == 2`, and `c == nil`. iow, you're explicitly setting `signing_cert` to `nil` and _implicitly_ setting `key` to `nil` because the rhs lacks enough objects to match the lhs.

Reply:
Also I think we should lift this to above the first `if`, incase the files don't exist we'll try to return uninitialized variables.